qemu/tests/virt_firmware_basic_test.py

```diff
@@ -0,0 +1,72 @@
+import os
+import re
+import shutil
+
+from avocado.utils import process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test check for virt-firmware.
+
+    The tests contain test-dump, test-vars,
+    test-sigdb, test-pe and test-unittest.
+
+    Clone the virt-firmware repo and run the following scripts.
+
+    test-dump:
+        tests/test-dump.sh
+    test-vars:
+        tests/test-vars.sh
+    test-sigdb:
+        tests/test-sigdb.sh
+    test-pe:
+        tests/test-pe.sh
+    test-unittest:
+        python3 tests/tests.py
+
+    :param test: VT test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    check_virt_firmware_cmd = params["check_virt_firmware_cmd"]
+    status = process.system(check_virt_firmware_cmd,
+                            ignore_status=True,
+                            shell=True)
+    if status:
+        test.cancel("Please install python-virt-firmware on the host.")
+    virt_firmware_dirname = params["virt_firmware_dirname"]
+    virt_firmware_repo_addr = params["virt_firmware_test_repo"]
+    clone_cmd = params["clone_cmd"]
+    test_file_pattern = params["test_file_pattern"]
+    if os.path.exists(virt_firmware_dirname):
+        shutil.rmtree(virt_firmware_dirname, ignore_errors=True)
+    clone_cmd %= (virt_firmware_repo_addr, virt_firmware_dirname)
+    status, output = process.getstatusoutput(clone_cmd,
+                                             ignore_status=True,
+                                             shell=True)
+    if status:
+        test.error("Failed to clone the virt-firmware repo,"
+                   "the error message is '%s'." % output)
+    test_dirname = os.path.join(virt_firmware_dirname, "tests")
+    for file_name in os.listdir(test_dirname):
+        if re.search(test_file_pattern, file_name, re.I):
+            test_file = os.path.join(test_dirname, file_name)
+            if file_name.endswith("py"):
+                test_cmd = params["python_cmd"] % test_file
```

Comment:
Suggest using `virttest.utils_test.find_python` to get the python bin.

Reply:
Thanks Yanan. This funciton is used for look for the python binary installed in the guest. For this case, run the tests in host.

Reply:
Yeah, you are right.   Then we can use `command -v python python3 | head -1` for it.

Reply:
Thanks Yanan, updated.

Reply:
`sys.executable`

Reply:
Thanks Yihuang, I will update it.